File: UI/src/app/models/request/requestLibrary.ts
// Library / watched-folder request payloads — mirrors
// UI/src-tauri/src/models/request/request_library.rs.

export interface requestRemoveFolder {
  path: string;
  purge: boolean;
}

export interface requestSetPaused {
  path: string;
  paused: boolean;
}

File: UI/src-tauri/src/models/request/request_library.rs
// Library / watched-folder request payloads.
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestRemoveFolder {
    pub path: String,
    pub purge: bool,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestSetPaused {
    pub path: String,
    pub paused: bool,
}
